References
==========

- Anders S., et al. `HTSeq — A Python framework to work with high-throughput
  sequencing data. <http://www-huber.embl.de/users/anders/HTSeq/doc/count.html>`_
  bioRxiv preprint. 2014
- Aronesty E. ea-utils: Command-line tools for processing biological
  sequencing data. Expression Analysis, Durham, NC. 2011
- Bao R., et al. `Review of Current Methods, Applications, and Data Management
  for the Bioinformatics Analysis of Whole Exome Sequencing.
  <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4179624/>`_ Cancer Informatics.
  2014; 13:67–82
- Bray N.L., et al. `Near-optimal probabilistic RNA-seq quantification. <http://www.nature.com/nbt/journal/v34/n5/abs/nbt.3519.html>`_
  Nature Biotechnology. 2016; 34:525–527
- Brennecke P., et al. `Accounting for technical noise in single-cell RNA-seq
  experiments. <http://www.natue.com/nmeth/journal/v10/n11/abs/nmeth.2645.html>`_
  Nature Methods. 2013; 10(11): 1093–1095
- Caporaso J.G. `QIIME allows analysis of high-throughput community sequencing
  data. <http://www.nature.comnmeth/journal/v7/n5/full/nmeth.f.303.html>`_
  Nature Methods. 2010; 7(5): 335-336
- `Cell Ontology <https://bioportal.bioontology.org/ontologies/CL>`_
- `Cellosaurus <http://web.expasy.org/cellosaurus/description.html>`_
- `The NCBI Taxonomy database <https://www.ncbi.nlm.nih.gov/taxonomy>`_
- `ChEBI <https://www.ebi.ac.uk/chebi/>`_
- Cingolani P. et al. `A program for annotating and predicting the effects of
  single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila
  melanogaster strin w1118; iso-2; iso-3. <http://www.tandfonline.com/doi/abs/10.4161/fly.19695>`_
  Fly. 2012; 6(2): 80-92
- Cokus S.J. `Shotgun bisulphite sequencing of the Arabidopsis genome reveals
  DNA methylation patterning. <http://www.nature.com/nature/journal/v452/n7184/abs/nature06745.html>`_
  Nature. 2008; 452(7184):215–219
- `CRAN <http://cran.r-project.org/>`_
- Davis M.P., et al. `Kraken: a set of tools for quality control and analysis
  of high-throughput sequence data.
  <http://www.sciencedirect.com/science/article/pii/S1046202313002399>`_
  Methods. 2013; 63(1):41-9
- Dobin A., et al. `STAR: ultrafast universal RNA-seq aligner.
  <https://academic.oup.com/biinformatics/article/29/1/15/272537/STAR-ultrafast-universal-RNA-seq-aligner>`_
  Bioinformatics. 2012; 29(1): 15-21
- Do C.B. and Batzoglou S. `What is the expectation maximization algorithm? <http://search.proquest.com/openview/f9d693573567370a6bd294384caf82d1/1?pq-origsite=gscholar&cbl=47191>`_
  Nature Biotechnology. 2008; 26(8):897-899
- Esposito D. `Cutting Edge - Programming CSS: Bundling and Minification.
  <https://msdn.microsoft.com/en-us/magazine/dn451436.aspx>`_ MSDN Magazine
  Blog. 2013
- Federhen S. `The NCBI Taxonomy database.
  <https://academic.oup.com/nar/article/40/D1/D136/2903327/The-NCBI-Taxonomy-database>`_
  Nucleic Acids Res. 2012
- Frankish A., et al. `Comparison of GENCODE and RefSeq gene annotation and
  the impact of reference geneset on variant effect prediction. <https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-16-S8-S2>`_
  BMC Genomics. 2015; 16(8), S2
- Gautier L. et al. `affy—analysis of Affymetrix GeneChip data at the probe
  level.
  <https://academic.oup.com/bioinformatics/article/20/3/307/185980/affy-analysis-of-Affymetrix-GeneChip-data-at-the>`_
  Bioinformatics. 2004; 20(3):307–315
- `Greengenes <http://greengenes.lbl.gov/cgi-bin/nph-index.cgi>`_
- Hastings J., et al. `The ChEBI reference database and ontology for
  biologically relevant chemistry: enhancements for 2013.
  <https://academic.oup.com/nar/article/41/D1/D456/1062873/The-ChEBI-reference-database-and-ontology-for>`_.
  Nucleic Acids Research. 2013
- `QIIME <http://qiime.org/1.2.1/index.html>`_
- Quinlan A.R. and Hall I.M. `BEDTools: a flexible suite of utilities for
  comparing genomic features. <https://academic.oup.com/bioinformatics/article/26/6/841/244688/BEDTools-a-flexible-suite-of-utilities-for>`_
  Bioinformatics. 2010; 26(6):841-2
- Kim D., et al. `TopHat2: accurate alignment of transcriptomes in the presence
  of insertions, deletions and gene fusions. <https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36>`_
  Genome Biology. 2013; 14(4), R36
- Langmead B. and Salzberg S.L. `Fast gapped-read alignment with Bowtie 2. <http://www.nature.com/nmeth/journal/v9/n4/abs/nmeth.1923.html>`_
  Nature Methods. 2012; 9(4):357-9
- Li B. and Dewey C.N. `RSEM: accurate transcript quantification from RNA-Seq
  data with or without a reference genome. <https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323>`_
  BMC Bioinformatics. 2011; 12(1), 323
- Li H. and Durbin R., `Fast and accurate short read alignment with
  Burrows-Wheeler Transform. <https://academic.oup.com/bioinformatics/article/25/14/1754/225615/Fast-and-accurate-short-read-alignment-with>`_
  Bioinformatics. 2009; 25:1754-60
- Li H., et al. `The Sequence Aignment/Map format and SAMtools. <https://academic.oup.com/bioinformatics/article/25/16/2078/204688/The-Sequence-Alignment-Map-format-and-SAMtools>`_
  Bioinformatics. 2009; 25(16): 2078-9
- Lister R., et al. `Human DNA methylomes at base resolution show widespread
  epigenomic differences. <http://www.nature.com/nature/journal/v462/n7271/abs/nature08514.html>`_
  Nature. 2009; 462(7271):315-22
- Love M.I., et al. `Moderated estimation of fold change and dispersion for
  RNA-seq data with DESeq2. <https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8>`_
  Genome Biology. 2014; 15(12), 550
- Luscombe N.M., et al. `What is bioinformatics? An introduction and overview.
  <https://www.ebi.ac.uk/luscombe/docs/imia_review.pdf>`_ Yearbook of Medical
  Informatics. 2001
- Madan Babu M.M. `An Introduction to Microarray Data Analysis
  <http://www.mrc-lmb.cam.ac.uk/genomes/madanm/microarray/chapter-final.pdf>`_
  Computational Genomics (Ed: R. Grant), Horizon press, UK 2004
- McCarthy D.J., et al. `Choice of transcripts and software has a large effect
  on variant annotation. <https://genomemedicine.biomedcentral.com/articles/10.1186/gm543>`_
  Genome Medicine. 2014; 6(3), 26
- Pearson W.R., et al. `Comparison of DNA sequences with protein sequences.
  <http://www.sciencedirect.com/science/article/pii/S0888754397949958>`_
  Genomics. 1997; 46(1):24-36
- `Removing duplicates is it really necessary?
  <http://seqanswers.com/forums/showthread.php?t=6854>`_ (2010, September 14) SEQanswers.
- Ritchie M.E., et al. `limma powers differential expression analyses for
  RNA-sequencing and microarray studies. <https://academic.oup.com/nar/article/43/7/e47/2414268/limma-powers-differential-expression-analyses-for>`_
  Nucleic Acids Research. 2015; 43(7)
- Robinson M.D., et al. `edgeR: a Bioconductor package for differential
  expression analysis of digital gene expression data. <https://academic.oup.com/bioinformatics/article/26/1/139/182458/edgeR-a-Bioconductor-package-for-differential>`_
  Bioinformatics. 2010; 26(1):139-140
- Trapnell C., et al. `Transcript assembly and quantification by RNA-Seq
  reveals unannotated transcripts and isoform switching during cell
  differentiation. <http://www.nature.com/nbt/journal/v28/n5/abs/nbt.1621.html>`_
  Nature Biotechnology. 2010; 28:511-515
- `UNITE <http://www2.dpes.gu.se/project/unite/UNITE_intro.htm>`_
- Xi Y. and Li W. `BSMAP: whole genome bisulfite sequence MAPping program. <https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-232>`_
  BMC Bioinformatics. 2009; 10(1), 232