References
==========

Introduction to bioinformatics
------------------------------

- Luscombe NM et al. "What is bioinformatics? An introduction and overview." Yearbook of Medical Informatics. 2001. (https://www.ebi.ac.uk/luscombe/docs/imia\_review.pdf)

Application management
----------------------

- https://msdn.microsoft.com/en-us/magazine/dn451436.aspx

Introduction to NGS
-------------------

- Riyue Bao, Lei Huang, Jorge Andrade, Wei Tan, Warren A Kibbe, Hongmei Jiang, Gang Feng. "Review of Current Methods, Applications, and Data Management for the Bioinformatics Analysis of Whole Exome Sequencing." Cancer Inform. 2014; 13(Suppl 2): 67–82. (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4179624)

Metadata
--------

- http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245000/ 
- http://web.expasy.org/cellosaurus/description.html
- https://www.ebi.ac.uk/chebi/
- https://bioportal.bioontology.org/ontologies/CL

Quality Control
---------------

- FastQC: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
- Aronesty E., “ea-utils: Command-line tools for processing biological sequencing data”, 2011; https://code.google.com/archive/p/ea-utils/
- Pearson W. R., Wood T. et al, “Comparison of DNA sequences with protein sequences.”, Genomics 1997, 46(1):24-36;  http://hannonlab.cshl.edu/fastx\_toolkit/ 
- Li H. “Toolkit for processing sequences in FASTA/Q formats."; https://github.com/lh3/seqtk
- Davis M.P., van Dongen S. et al, “Kraken: a set of tools for quality control and analysis of high-throughput sequence data.”, Methods 2013, 63(1):41-9;  http://www.ebi.ac.uk/~stijn/reaper/tally.html
- http://seqanswers.com/forums/showthread.php?t=6854

Choosing a mapper
-----------------

- Kim D, Pertea G, Trapnell C, Pimentel H, Kelley R, Salzberg SL. "TopHat2: accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions.", Genome Biol. 2013 Apr 25;14(4):R36; http://ccb.jhu.edu/software/tophat/index.shtml
- Li H. and Durbin R., “Fast and accurate short read alignment with Burrows-Wheeler Transform”, Bioinformatics, 2009, 25:1754-60; http://bio-bwa.sourceforge.net/
- Langmead B, Salzberg SL. "Fast gapped-read alignment with Bowtie 2." Nat Methods. 2012 Mar 4;9(4):357-9. doi: 10.1038/nmeth.1923.; http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml
- Xi Y. and Li W., "BSMAP: whole genome bisulfite sequence MAPping program.", BMC Bioinformatics. 2009 10:232. doi: 10.1186/1471-2105-10-232
https://code.google.com/archive/p/bsmap/
- Lister R, Pelizzola M, Dowen RH, Hawkins RD, Hon G, Tonti-Filippini J, Nery JR, Lee L, Ye Z, Ngo Q-M, Edsall L, Antosiewicz-Bourget J, Stewart R, Ruotti V, Millar AH, Thomson JA, Ren B, Ecker JR. "Human DNA methylomes at base resolution show widespread epigenomic differences." Nature. 2009 462(7271):315-22. (http://europepmc.org/abstract/MED/19829295)
- Cokus SJ, Feng S, Zhang X, Chen Z, Merriman B, Haudenschild CD, Pradhan S, Nelson SF, Pellegrini M, Jacobsen SE. "Shotgun bisulphite sequencing of the Arabidopsis genome reveals DNA methylation patterning." Nature. 2008 452(7184):215–219. (http://europepmc.org/abstract/MED/18278030)

Reference genomes
-----------------

- McCarthy DJ, Humburg P, Kanapin A, Rivas MA, Gaulton K, Cazier JB, Donnelly P. Choice of transcripts and software has a large effect on variant annotation. Genome Med. 2014;6(3):26 (https://genomemedicine.biomedcentral.com/articles/10.1186/gm543)
- Frankish A, Uszczynska B, Ritchie GR, Gonzalez JM, Pervouchine D, Petryszak R, et al. Comparison of GENCODE and RefSeq gene annotation and the impact of reference geneset on variant effect prediction. BMC Genomics. 2015;16 (Suppl 8):S2 (http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-16-S8-S2)

Microarray Normalization
------------------------

- Madan Babu M.M. "An Introduction to Microarray Data Analysis" Computational Genomics (Ed: R. Grant), Horizon press, UK 2004 (http://www.mrc-lmb.cam.ac.uk/genomes/madanm/microarray/chapter-final.pdf)
- Gautier L, Cope L, Bolstad BM and Irizarry RA (2004). “affy—analysis of Affymetrix GeneChip data at the probe level.” Bioinformatics, 20(3), pp. 307–315; https://www.bioconductor.org/packages/3.3/bioc/html/affy.html
- Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W and Smyth GK (2015). “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), pp. e47;
https://www.bioconductor.org/packages/3.3/bioc/html/limma.html
- Gautier L. et al. "affy—analysis of Affymetrix GeneChip data at the probe level." 2004 Bioinformatics, 20(3), pp. 307–315 (http://www.ncbi.nlm.nih.gov/pubmed/14960456)
- Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W and Smyth GK "limma powers differential expression analyses for RNA-sequencing and microarray studies." 2015 Nucleic Acids Research, 43 (7), pp. e47 (http://europepmc.org/abstract/MED/25605792)

Preformating tools
------------------

- BCFtools: http://samtools.github.io/bcftools/bcftools.html
- Samtools: http://samtools.sourceforge.net/
- Picard: http://broadinstitute.github.io/picard/

Analysis tools
--------------

- Anders S, Pyl PT, Huber W. "HTSeq — A Python framework to work with high-throughput sequencing data." bioRxiv preprint. 2014; http://www-huber.embl.de/users/anders/HTSeq/doc/count.html

Visualization tools
-------------------

- PaGE: http://www.cbil.upenn.edu/PaGE/fdr.html


