References
==========

Introduction to bioinformatics
------------------------------

- Luscombe NM et al. "What is bioinformatics? An introduction and overview." Yearbook of Medical Informatics. 2001; https://www.ebi.ac.uk/luscombe/docs/imia\_review.pdf

Application management
----------------------

- Esposito D. "Cutting Edge - Programming CSS: Bundling and Minification" (2013, October) MSDN Magazine Blog. Retrieved from: https://msdn.microsoft.com/en-us/magazine/dn451436.aspx


Introduction to NGS
-------------------

- Riyue Bao, Lei Huang, Jorge Andrade, Wei Tan, Warren A Kibbe, Hongmei Jiang, Gang Feng. "Review of Current Methods, Applications, and Data Management for the Bioinformatics Analysis of Whole Exome Sequencing." Cancer Inform. 2014; 13(Suppl 2): 67–82; https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4179624

Metadata
--------

- Cell Ontology: https://bioportal.bioontology.org/ontologies/CL
- Cellosaurus: http://web.expasy.org/cellosaurus/description.html
- Federhen S. "The NCBI Taxonomy database." Nucleic Acids Res. 2012 Jan; 40(Database issue): D136–D143. 
- Hastings, J. et al. (2013) The ChEBI reference database and ontology for biologically relevant chemistry: enhancements for 2013. Nucleic Acids Res. https://www.ebi.ac.uk/chebi/


Quality Control and Preprocessing
---------------------------------

- Aronesty E., “ea-utils: Command-line tools for processing biological sequencing data”, 2011. This tool is available at https://code.google.com/archive/p/ea-utils/
- Davis M.P., van Dongen S. et al, “Kraken: a set of tools for quality control and analysis of high-throughput sequence data.”, Methods 2013, 63(1):41-9
- FastQC: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
- FASTX-Toolkit: http://hannonlab.cshl.edu/fastx\_toolkit/
- Gautier L, Cope L, Bolstad BM and Irizarry RA (2004). “affy—analysis of Affymetrix GeneChip data at the probe level.” Bioinformatics, 20(3), pp. 307–315; https://www.bioconductor.org/packages/3.3/bioc/html/affy.html 
- Madan Babu M.M. "An Introduction to Microarray Data Analysis" Computational Genomics (Ed: R. Grant), Horizon press, UK 2004; http://www.mrc-lmb.cam.ac.uk/genomes/madanm/microarray/chapter-final.pdf
- Pearson W. R., Wood T. et al, “Comparison of DNA sequences with protein sequences.”, Genomics 1997, 46(1):24-36
- "Removing duplicates is it really necessary?" (2010, September 14) SEQanswers. Retrieved from: http://seqanswers.com/forums/showthread.php?t=6854
- Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W and Smyth GK (2015). “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), pp. e47; https://www.bioconductor.org/packages/3.3/bioc/html/limma.html
- Seqtk: https://github.com/lh3/seqtk
- Tally: http://www.ebi.ac.uk/~stijn/reaper/tally.html

Preformating tools
------------------

- BCFtools: http://samtools.github.io/bcftools/bcftools.html
- Li H et al. "The Sequence Alignment/Map format and SAMtools." Bioinformatics. 2009 Aug 15;25(16):2078-9.; http://samtools.sourceforge.net/
- Picard: http://broadinstitute.github.io/picard/

Analysis tools
--------------

- Anders S, Pyl PT, Huber W. "HTSeq — A Python framework to work with high-throughput sequencing data." bioRxiv preprint. 2014; http://www-huber.embl.de/users/anders/HTSeq/doc/count.html
- Bioconductor package limma: https://www.bioconductor.org/packages/release/bioc/html/limma.html
- Bray N L, Pimentel H, Melsted P and Pachter L. “Near-optimal probabilistic RNA-seq quantification.” Nature Biotechnology 2016 34:525–527; https://pachterlab.github.io/kallisto/
- Brennecke P, Anders S, Kim JK, Kolodziejczyk AA, Zhang X, Proserpio V, Baying B, Benes V, Teichmann SA, Marioni JC, Heisler MG. “Accounting for technical noise in single-cell RNA-seq experiments.” Nature Methods. 2013 Sep 22; 10(11):1093–1095.
- Caporaso J. G "QIIME allows analysis of high-throughput community sequencing data” Nature Methods, 2010.
- Cingolani P, Platts A, Wang le L, Coon M, Nguyen T, Wang L, Land SJ, Lu X, Ruden DM. "A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3." Fly (Austin). 2012 Apr-Jun;6(2):80-92.;  http://snpeff.sourceforge.net/SnpEff_manual.html
- Cokus SJ, Feng S, Zhang X, Chen Z, Merriman B, Haudenschild CD, Pradhan S, Nelson SF, Pellegrini M, Jacobsen SE. "Shotgun bisulphite sequencing of the Arabidopsis genome reveals DNA methylation patterning." Nature. 2008 452(7184):215–219; http://europepmc.org/abstract/MED/18278030
- CRAN: http://cran.r-project.org/
- Dobin A, Davis C A, Schlesinger F, Drenkow J, Zaleski C, Jha S, Batut P, Chaisson M and Gingeras T R. “STAR: ultrafast universal RNA-seq aligner.” Bioinformatics 2012 29(1): 15-21; https://github.com/alexdobin/STAR
- Do C B and Batzoglou S. “What is the expectation maximization algorithm?” Nature biotechnology, 2008 26(8): 897-899.
- Greengenes: http://greengenes.lbl.gov/cgi-bin/nph-index.cgi
- QIIME: http://qiime.org/1.2.1/index.html
- Quinlan AR, Hall IM. “BEDTools: a flexible suite of utilities for comparing genomic features.” Bioinformatics. 2010 Mar 15;26(6):841-2.; http://bedtools.readthedocs.org/en/latest/content/tools/intersect.html
- Kim D, Pertea G, Trapnell C, Pimentel H, Kelley R, Salzberg SL. "TopHat2: accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions.", Genome Biol. 2013 Apr 25;14(4):R36; http://ccb.jhu.edu/software/tophat/index.shtml
- Langmead B, Salzberg SL. "Fast gapped-read alignment with Bowtie 2." Nat Methods. 2012 Mar 4;9(4):357-9. doi: 10.1038/nmeth.1923.; http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml
- Li B and Dewey C N. “RSEM: accurate transcript quantification from RNA-Seq data with or without a reference genome.” BMC Bioinformatics 2011 12:323; http://deweylab.github.io/RSEM/
- Li H. and Durbin R., “Fast and accurate short read alignment with Burrows-Wheeler Transform”, Bioinformatics, 2009, 25:1754-60; http://bio-bwa.sourceforge.net/
- Lister R, Pelizzola M, Dowen RH, Hawkins RD, Hon G, Tonti-Filippini J, Nery JR, Lee L, Ye Z, Ngo Q-M, Edsall L, Antosiewicz-Bourget J, Stewart R, Ruotti V, Millar AH, Thomson JA, Ren B, Ecker JR. "Human DNA methylomes at base resolution show widespread epigenomic differences." Nature. 2009 462(7271):315-22; http://europepmc.org/abstract/MED/19829295
- Love MI, Huber W and Anders S. “Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.” Genome Biology. 2014; 15(12):550.
- Robinson MD, McCarthy DJ and Smyth GK. “edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.” Bioinformatics. 2010; 26(1):139-140.
- "The False Discovery Rate".  Retrieved from: http://www.cbil.upenn.edu/PaGE/fdr.html
- Trapnell C, Williams BA, Pertea G, Mortazavi A, Kwan G, van Baren MJ, Salzberg SL, Wold BJ, Pachter L. "Transcript assembly and quantification by RNA-Seq reveals unannotated transcripts and isoform switching during cell differentiation.", Nature Biotechnology. 2010 May 02; 28:511-515; http://cufflinks.cbcb.umd.edu/
- UNITE: http://www2.dpes.gu.se/project/unite/UNITE_intro.htm
- Xi Y. and Li W., "BSMAP: whole genome bisulfite sequence MAPping program.", BMC Bioinformatics. 2009 10:232. doi: 10.1186/1471-2105-10-232; https://code.google.com/archive/p/bsmap/


Reference genomes
-----------------

- Frankish A, Uszczynska B, Ritchie GR, Gonzalez JM, Pervouchine D, Petryszak R, et al. Comparison of GENCODE and RefSeq gene annotation and the impact of reference geneset on variant effect prediction. BMC Genomics. 2015;16 (Suppl 8):S2; http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-16-S8-S2
- McCarthy DJ, Humburg P, Kanapin A, Rivas MA, Gaulton K, Cazier JB, Donnelly P. Choice of transcripts and software has a large effect on variant annotation. Genome Med. 2014;6(3):26; https://genomemedicine.biomedcentral.com/articles/10.1186/gm543

